app/services/link_check_report/create_service.rb

```diff
@@ -0,0 +1,77 @@
+require "services"
+require "govspeak/link_extractor"
+
+class LinkCheckReport::CreateService
+  class InvalidReport < RuntimeError
+    def initialize(original_error)
+      @message = original_error.message
+    end
+  end
+
+  def initialize(user:, manual_id:, section_id: nil)
+    @user = user
+    @manual_id = manual_id
+    @section_id = section_id
+  end
+
+  def call
+    link_report = call_link_checker_api
+
+    report = LinkCheckReport.new(
+      batch_id: link_report.fetch(:batch_id),
+      completed_at: link_report.fetch(:completed_at),
+      status: link_report.fetch(:status),
+      manual_id: manual_id,
+      section_id: section_id,
+      links: link_report.fetch(:links).map { |link| map_link_attrs(link) }
+    )
+
+    report.save!
+  rescue Mongoid::Errors::Validations => e
+    raise InvalidReport.new(e)
+  end
+
+private
+
+  attr_reader :user, :manual_id, :section_id
+
+  def manual
+    @manual ||= Manual.find(manual_id, user)
+  end
+
+  def section
```

Comment:
We could add a `raise "Not a section" if section_id.nil?`.

Reply:
Hmm just thinking perhaps we should use `unless is_for_section?` here instead to avoid duplicating logic.